_posts/2022-09-29-r-text/r-text.Rmd

```diff
@@ -0,0 +1,151 @@
+---
+title: "The text-package"
+description: |
+  The [text](http://r-text.org/)-package attempts to provide user-friendly access and pipelines to HuggingFace's transformer language models in R.
+author:
+  - name: Oscar Kjell 
+    url: https://oscarkjell.se/
+    affiliation: Lund University
+    affiliation_url: https://portal.research.lu.se/en/persons/oscar-kjell
+  - name: Salvatore Giorgi 
+    url: http://www.salvatoregiorgi.com/
+    affiliation: University of Pennsylvania
+    affiliation_url: http://www.salvatoregiorgi.com/
+  - name: H Andrew Schwartz  
+    url: https://www3.cs.stonybrook.edu/~has/
+    affiliation: Stony Brook University
+    affiliation_url: https://www3.cs.stonybrook.edu/~has/
+categories:
+  - Natural Language Processing
+  - Machine Learning
+  - Transformers
+  - HuggingFace
+date: 09-29-2022
+creative_commons: CC BY
+repository_url: https://github.com/OscarKjell/ai-blog
+output: distill::distill_article
+preview: img/ai_blog_word_plot.png
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = FALSE,
+                      eval = FALSE)
+```
+
+AI-based language analysis has recently gone through a “paradigm shift” (Bommasani, et al., 2021, p. 1) thanks in part to a new technique referred to as transformer language model (Vaswani et al., 2017, Liu et al., 2019). Companies, including Google, Meta, and OpenAI have released such models including BERT, RoBERTa, and GPT, that have achieved unprecedented large improvements across most language tasks such as web search and sentiment analysis. While these language models are accessible in Python, and for typical AI tasks through [HuggingFace](https://huggingface.co/), the R-package [text](http://r-text.org/) makes HuggingFace and state-of-the-art transformer language models accessible as social scientific pipelines in R.
+
+
+## Introduction
+
+We developed the `text`-package [(Kjell, Giorgi & Schwartz, 2022)](https://psyarxiv.com/293kt) with two objectives in mind: 
+To serve as a modular solution for downloading and using transformer language models. This, for example, includes transforming text to word embeddings as well as accessing common language model tasks such as text classification, sentiment analysis, text generation, question answering, translation and so on. 
+To provide an end-to-end solution that is catered for human-level analyses including pipelines for state-of-the-art AI techniques tailored for predicting characteristics of the person that produced the language or eliciting insights about linguistic correlates of psychological attributes.
+
+This blog post shows how to install the `text`-package, transforming text to state-of-the-art contextual word embeddings, using language analysis tasks as well as visualizing words in word embedding space.  
```

Comment:
This blog post shows how to install the `text`-package, transforming text to state-of-the-art contextual word embeddings, using language analysis tasks as well as visualizing words in word embedding space.

This blog post shows how to install the `text` package, transform text to state-of-the-art contextual word embeddings, use language analysis tasks as well as visualize words in word embedding space.